group/edwards25519/curve.go

```diff
@@ -51,6 +51,5 @@ func (c *Curve) NewKey(stream cipher.Stream) kyber.Scalar {
 	scalar[31] &= 0x3f
 	scalar[31] |= 0x40
 
-	secret := c.Scalar().SetBytes(scalar[:32])
-	return secret
+	return c.Scalar().SetBytes(scalar[:32])
```

Comment:
In fact here we want a non-reduced one, secret key is never modified "as-is" in the RFC.